pkg/controller/controller_instance.go

```diff
@@ -2218,6 +2234,46 @@ func (c *controller) checkServiceInstanceHasExistingBindings(instance *v1beta1.S
 	return nil
 }
 
+func (c *controller) listExistingBindings(instance *v1beta1.ServiceInstance) ([]*v1beta1.ServiceBinding, error) {
+	bindingLister := c.bindingLister.ServiceBindings(instance.Namespace)
+
+	selector := labels.NewSelector()
+	bindingList, err := bindingLister.List(selector)
+	if err != nil {
+		return []*v1beta1.ServiceBinding{}, err
+	}
+	var found []*v1beta1.ServiceBinding
+	for _, binding := range bindingList {
+		// Note that as we are potentially looking at a stale binding resource
+		// and cannot rely on UnbindStatus == ServiceBindingUnbindStatusNotRequired
+		// to filter out binding requests that have yet to be sent to the broker.
+		if instance.Name == binding.Spec.InstanceRef.Name {
+			found = append(found, binding)
+		}
+	}
+
+	return found, nil
+}
+
+func (c *controller) deleteExistingBindings(instance *v1beta1.ServiceInstance) error {
+	klog.V(4).Infof("Delete existing bindings for the instance %s", instance.Name)
+	bindings, err := c.listExistingBindings(instance)
+	if err != nil {
+		return err
```

Comment:
u're missing context for each error
e.g. here u should have 

```go
errors.Wrapf(err, "while fetching ServiceBindings for instance %s", instance.Name) 
```

Reply:
I've added wrapping errors